static/app/views/performance/trends/changeExplorer.tsx

```diff
@@ -0,0 +1,815 @@
+  TrendParameter,
+  TrendsStats,
+  TrendsTransaction,
+  TrendView,
+} from 'sentry/views/performance/trends/types';
+import SlideOverPanel from 'sentry/views/starfish/components/slideOverPanel';
+
+type PerformanceChangeExplorerProps = {
+  collapsed: boolean;
+  isLoading: boolean;
+  location: Location;
+  onClose: () => void;
+  organization: Organization;
+  projects: Project[];
+  statsData: TrendsStats;
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+  trendFunction: string;
+  trendParameter: TrendParameter;
+  trendView: TrendView;
+};
+
+type ExplorerBodyProps = {
+  isLoading: boolean;
+  location: Location;
+  organization: Organization;
+  projects: Project[];
+  statsData: TrendsStats;
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+  trendFunction: string;
+  trendParameter: TrendParameter;
+  trendView: TrendView;
+};
+
+type HeaderProps = {
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+};
+
+type MetricsChartProps = {
+  isLoading: boolean;
+  location: Location;
+  organization: Organization;
+  transaction: NormalizedTrendsTransaction;
+  trendFunction: string;
+  trendView: TrendView;
+};
+
+export function PerformanceChangeExplorer({
+  collapsed,
+  transaction,
+  onClose,
+  trendChangeType,
+  trendFunction,
+  trendView,
+  statsData,
+  isLoading,
+  organization,
+  projects,
+  trendParameter,
+  location,
+}: PerformanceChangeExplorerProps) {
+  const panelRef = useRef<HTMLDivElement>(null);
+  useOnClickOutside(panelRef, () => {
+    if (!collapsed) {
+      onClose();
+    }
+  });
+
+  const escapeKeyPressed = useKeyPress('Escape');
+
+  useEffect(() => {
+    if (escapeKeyPressed) {
+      if (!collapsed) {
+        onClose();
+      }
+    }
+  }, [escapeKeyPressed, collapsed, onClose]);
+
+  return (
+    <SlideOverPanel collapsed={collapsed} ref={panelRef}>
+      <CloseButtonWrapper>
+        <CloseButton
+          priority="link"
+          size="zero"
+          borderless
+          aria-label={t('Close Details')}
+          icon={<IconClose size="sm" />}
+          onClick={onClose}
+        />
+      </CloseButtonWrapper>
+      <PanelBodyWrapper>
+        <ExplorerBody
+          transaction={transaction}
+          trendChangeType={trendChangeType}
+          trendFunction={trendFunction}
+          trendView={trendView}
+          statsData={statsData}
+          isLoading={isLoading}
+          organization={organization}
+          projects={projects}
+          trendParameter={trendParameter}
+          location={location}
+        />
+      </PanelBodyWrapper>
+    </SlideOverPanel>
+  );
+}
+
+function ExplorerBody(props: ExplorerBodyProps) {
+  const {
+    transaction,
+    trendChangeType,
+    trendFunction,
+    trendView,
+    trendParameter,
+    isLoading,
+    location,
+    organization,
+  } = props;
+  const breakpointDate = new Date(transaction.breakpoint! * 1000)
```

Comment:
`breakpoint` is optional on `transaction` for backward compatibility with old trends since no breakpoints will be sent then. but for the trend explorer that is only available for new trends, it's safe to assume that breakpoint will not be undefined